### PR TITLE
Full localization in Open Shell

### PR DESCRIPTION
**Pull request by vodek3**
HTML Title

LocImmersive

### CODE REVIEW

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc [hunk 1]
```diff
@@ -1181,6 +1188,8 @@ BEGIN
     IDS_STRING7108          "Midnight skin\n\nSkin with dark background.\n\nPart of <A HREF=""https://github.com/Open-Shell/Open-Shell-Menu/"">Open-Shell</A> (c) 2009-2017, Ivo Beltchev"
     IDS_STRING7109          "Metro skin\n\nSkin that uses the start screen colors.\n\nPart of <A HREF=""https://github.com/Open-Shell/Open-Shell-Menu/"">Open-Shell</A> (c) 2009-2017, Ivo Beltchev"
     IDS_STRING7110          "Metallic skin\n\nA start menu skin with metallic look.\n\nPart of <A HREF=""https://github.com/Open-Shell/Open-Shell-Menu/"">Open-Shell</A> (c) 2009-2017, Ivo Beltchev"
+    IDS_STRING7111          "Immersive skin\n\nA skin that uses immersive colors and modern visual elements in the style of Windows 10.\n\nPart of <A HREF=""https://github.com/Open-Shell/Open-Shell-Menu/"">Open-Shell</A> (c) 2009-2017, Ivo Beltchev"
+
```
Comment (ge0rdi) [inline Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:1192]: Unnecessary blank line.

**Review by ge0rdi — CHANGES_REQUESTED** [review]

**Comment by ge0rdi** [comment]
Regarding `Add to Help two chapters + Changes of formatting in head of all .htm files` commit.
I'm wondering what was the problem with having each change in separate commit.

We were discussing in last PR that unrelated changes should go to separate commits.

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc [hunk 2]
```diff
@@ -932,7 +932,7 @@ BEGIN
     IDS_SEARCH_KEYWORDS_TIP "When this is checked, the search will look at the keywords associated with the settings"
     IDS_SEARCH_FILES        "Search files"
     IDS_SEARCH_FILES_TIP    "When this is checked, the search results will include files, emails and other items from indexed locations"
-    IDS_SEARCH_FILES_TIP2   "When this is checked, the search results will include files, emails  and other items from indexed locations\nWarning: the search service is disabled"
+    IDS_SEARCH_FILES_TIP2   "When this is checked, the search results will include files, emails and other items from indexed locations\nWarning: the search service is disabled"
```
[the remark this answers is not in the document]
Comment (vodek3) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: Two spaces in this string also connected with full localization of Open-Shell, I'll write about it bit later.
I have a suspicion that this is a bug. Now I'm just checking it out.

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc (hunk 1 above)
Comment (vodek3) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc]: Agree

**Comment by vodek3** [comment]
> 
I answered above

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: While two spaces in this string don't make much sense, it makes even less sense to _fix_ that in commit that is about localization of `Immersive` skin.

So please just revert the change.

File: Src/Localization/English/OpenShell.hhp [hunk 3]
```diff
@@ -5,12 +5,67 @@ Contents file=OpenShellTOC.hhc
...
 Display compile progress=Yes
 Language=0x409 English (United States)
-
+Title=Open-Shell Help
 
 [FILES]
 ClassicExplorer.html
 StartMenu.html
 ClassicIE.html
+ButtonTutorial.html
```
[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/Localization/English/OpenShell.hhp:14]: > They are inextricably linked

No.
Commit `Change title of Help Window: Help -> Open-Shell Help` is supposed to just change the title of help window.
Adding some other things is certainly not related to help window title.
So those other things should go to other commit - `Add to Help two chapters` that adds those help chapters.

**Comment by ge0rdi** [comment]
> Remove blank line

Here is an advice for creating _fixup_ commits like this one.
You can create the commit using `git commit --fixup <id-of-commit-that-is-being-fixed>`.

It will be then much easier to squash particular fixups in to their base commits once review is finished.
One can then use `git rebase -i --autosquash ...` and git will automatically apply fixup commits to their base commits.

File: Src/Localization/English/OpenShell.hhp (hunk 3 above)
[the remark this answers is not in the document]
Comment (vodek3) [reply Src/Localization/English/OpenShell.hhp:14]: Listen, what are we talking about?
I made a separate commit to change the title of Help Windows.
Only by changing .hpp files
Maybe I didn't understand you correctly?
image

[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/Localization/English/OpenShell.hhp:14]: > I made a separate commit to change the title of Help Windows.
> Only by changing .hpp files

That is right. But some of those `.hpp` files contain also other changes than just change in title.
Just have a look at diff of that commit:
https://github.com/Open-Shell/Open-Shell-Menu/pull/1317/commits/114369899ad9f3a789f5ab1d3e6d11041c3b0350

So you need to put those change so different commit (I guess they mixed in somehow).

[the remark this answers is not in the document]
Comment (vodek3) [reply Src/Localization/English/OpenShell.hhp:14]: I named the commit "Add two chapters to help + Format header changes in all .html" because they are inextricably linked! And also all the changes are connected with the name of the PR - full localization.
It is absurd to create a commit after deleting each character I underline in the, not in the code, but in the documentation. By the way, I created a revert on two spaces in resource-file, as you asked.

[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/Localization/English/OpenShell.hhp:14]: > I named the commit "Add two chapters to help + Format header changes in all .html" because they are inextricably linked!

And I'm talking about commit `Change title of Help Window: Help -> Open-Shell Help` (the one where I left this comment): 
https://github.com/Open-Shell/Open-Shell-Menu/pull/1317/commits/114369899ad9f3a789f5ab1d3e6d11041c3b0350

Just click on it, see the differences made in that commit and in `Src/Localization/English/OpenShell.hhp` and `Src/Localization/Russian/OpenShell.hhp` files you will see changes that are not related to change of help title.

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc (hunk 2 above)
[the remark this answers is not in the document]
Comment (vodek3) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: I described bug in #1318

**Comment by ge0rdi** [comment]
Could you, please, squash fixes so that in the end there will be just commits you want to merge in this PR?
You can do that locally (look for "git interactive rebase") and then once you are satisfied with it, just force push it (`git push -f`).

So that it will be easier to review actual changes.

**Comment by ge0rdi** [comment]
For the future, it is much better to prepare all the commits locally and then push them at once.
Instead of pushing one commit here and there.

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: I thought it was already explained that this change doesn't belong to commit that is about Immersive skin changes.
Moreover it is not relate to #1318 in any way.

File: Src/Localization/English/ButtonTutorial.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
```
Comment (ge0rdi) [inline Src/Localization/English/ButtonTutorial.html:1]: New help pages should be added in separate commit as they are introducing new content.
It makes no sense to add completely new pages in commit that aims to make existing help files more uniform.

**Review by ge0rdi — CHANGES_REQUESTED** [review]
These commits still contain mix of different things. I've tried to explain it as best as I can,

How about this:
I have all required data here, so I will just rearrange them a bit to have more meaningful commit.
I can try to do it here (not sure if I can modify commits in PR where I'm not the author).
Or will create new PR with required changes (you will be listed as author of commits).

Will that work for you?

File: Src/Localization/English/ButtonTutorial.html (hunk 4 above)
Comment (vodek3) [reply Src/Localization/English/ButtonTutorial.html:1]: I repeat. My main goal is to make it possible to easily localize all Open Shell messages (including group policy templates).
By the way. I made (a few years ago) for myself the ru-RU.dll library and it works correctly. I don't use the Immersive skin. ;-)
But there is what is called the shift of the motive to the goal - it became interesting for me to finish what I started and share it with everyone.
And so.
I absolutely do not care who will be indicated by the author of the changes in the source code. Make it as easy as possible for you. The main thing is the result! 
If any action is required from me, please let me know.
Once again, I would like to note that all the proposed changes/improvements in this PR are directly related to localization.
It's important.

File: Src/StartMenu/StartMenuDLL/StartMenuDLL.rc (hunk 2 above)
[the remark this answers is not in the document]
Comment (vodek3) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: Displaying in the template (more precisely, not displaying) a message with two spaces is described in #1318 and discussion of the same message (with two spaces) is not related to #1318?
Really?
Do you seriously think that two spaces in the message are so important that we need to spend more than two days discussing this issue?..

Two days... We wasted two days. For what? For replace two spaces by one. This is a success! =D

[the remark this answers is not in the document]
Comment (ge0rdi) [reply Src/StartMenu/StartMenuDLL/StartMenuDLL.rc:935]: Hmm, I thought I explained it in #1318 already.
It was related to settings handling code that depends on state of particular Windows service. It is not related to number of spaces in some string.
Please, try to read it again.

That issue is fixed now. If you think it is not, please, state that in #1318.

**Comment by vodek3** [comment]
I explain my goal minimum twice...
Look, when compile for example Russian version of Open-Shell help compile automatically and an installation package is created, including a help file.  So it turns out that the localized help will differ from the English one without two chapters...
Let's start from revise and unify all fields for subsequent translation.  I think this is a very good idea because it is substantive and quantitative. 
After that we will remove all double spaces of course. ;)

**Comment by ge0rdi** [comment]
I understand what you wrote, but I don't agree with all of it.
I don't want any unnecessary localization changes, nor adding untranslated files to existing localizations.

I have prepared PR #1320 with changes that are imo OK.
So please have a look and leave some feedback there eventually.

Once that PR is in master, you can submit PR that will deal with `Russian` part of localizations.

I will close this PR as it is not relevant anymore.
